hastory-cli/test/Hastory/Cli/OptParse/TypesSpec.hs

```diff
@@ -33,8 +33,54 @@ spec =
              })
     it "is an error when user provides the wrong type for a field" $ do
       value <- Data.Yaml.decodeThrow "url: 1"
-      (fromJSON value :: Result Configuration) `shouldBe`
-        Error "expected String, but encountered Number"
+      (fromJSON value :: Result Configuration) `shouldSatisfy` isConfigParserError
+    context "cache-dir" $ do
+      it "parses correctly when cache-dir key is not provided" $ do
+        value <- Data.Yaml.decodeThrow "url: api.example.com"
+        let Success config = fromJSON value
+        configCacheDir config `shouldBe` Nothing
+      it "parses correctly when cache-dir key is provided and value is valid" $ do
+        value <- Data.Yaml.decodeThrow "cache-dir: ~/home"
```

Comment:
This will parse as a string correctly, but `~/` is expanded by your shell so it won't mean "home" when you use it as a string like this.

Reply:
Since the user can in type that into their configuration file, and we want to do as little processing as possible, do you still want to keep this? I believe we parse this to a stronger type in `combineToInstructions` later one, using the functions in `path-io` library.